source/FreeRTOS_DHCPv6.c

```diff
@@ -386,7 +386,7 @@ void vDHCPv6Process( BaseType_t xReset,
 static void vDHCPv6ProcessEndPoint_HandleReply( NetworkEndPoint_t * pxEndPoint,
                                                 DHCPMessage_IPv6_t * pxDHCPMessage )
 {
-    FreeRTOS_printf( ( "vDHCPProcess: acked %lxip\n", FreeRTOS_ntohl( EP_DHCPData.ulOfferedIPAddress ) ) );
+    FreeRTOS_printf( ( "vDHCPProcess: acked %x ip\n", ( unsigned int ) FreeRTOS_ntohl( EP_DHCPData.ulOfferedIPAddress ) ) );

```

Comment:
And this is perfect: a cast to `unsigned` and removing the l (long) prefix.
However, please do not separate the `x` from the `ip`. It is used to format the parameter as an IPv4 IP address.

~~~
    FreeRTOS_printf( ( "vDHCPProcess: acked %xip\n", ( unsigned int ) FreeRTOS_ntohl( EP_DHCPData.ulOfferedIPAddress ) ) );
~~~
The above may print e.g. `"vDHCPProcess: acked 192.168.2.5"`. See also [printf-stdarg.c](https://github.com/htibosch/freertos_plus_projects/blob/master/plus/Common/Utilities/printf-stdarg.c), which is also included in some of the demos.

IPv6 addresses can be printed as: 
~~~
    FreeRTOS_printf( ( "RA: source %pip\n", xSourceAddress.ucBytes ) );
~~~
which may result in `"RA: source fe80::7007"`.